app/wiring/AppComponents.scala

```diff
@@ -4,25 +4,32 @@ import com.gu.googleauth.AuthAction
 import com.gu.memsub.auth.common.MemSub.Google.googleAuthConfigFor
 import com.gu.memsub.promo.{Campaign, DynamoTables}
 import com.gu.memsub.promo.Promotion.AnyPromotion
-import play.api.BuiltInComponents
+import play.api.BuiltInComponentsFromContext
 import com.typesafe.config.{Config, ConfigFactory}
 import com.softwaremill.macwire._
-import conf.{CatalogService, PaperPlans, PaperProducts, WeeklyPlans}
+import com.typesafe.scalalogging.LazyLogging
+import conf.{CatalogService, PaperProducts, WeeklyPlans}
 import controllers._
+import play.api.ApplicationLoader.Context
 import play.api.libs.ws.ahc.AhcWSComponents
-import play.api.mvc.{AnyContent, ControllerComponents}
+import play.api.mvc.{AnyContent, EssentialFilter}
 import play.api.routing.Router
+import play.filters.HttpFiltersComponents
+import play.filters.csrf.CSRFFilter
+import play.filters.hosts.AllowedHostsFilter
 import wiring.AppComponents.Stage
 import router.Routes
 
-import scala.concurrent.ExecutionContext
+class AppComponents(context: Context) extends BuiltInComponentsFromContext(context) with AhcWSComponents with AssetsComponents with HttpFiltersComponents {
 
-class AppComponents(private val stage: Stage, c: BuiltInComponents with AhcWSComponents, controllerComponents: ControllerComponents, assetsComponents: AssetsComponents) {
-
-  import c._
-  import assetsComponents.assetsMetadata
+  override def httpFilters: Seq[EssentialFilter] =
```

Comment:
`httpFilters` was previously set in `AppLoader` but it makes more sense to set it in here now